kong/clustering/data_plane.lua

```diff
@@ -209,6 +210,17 @@ function _M:communicate(premature)
 
   local conf = self.conf
 
+  do
+    local response_data, err = version_negotiation.request_version_handshake(conf, self.cert, self.cert_key)
+    if not response_data then
+      ngx_log(ngx_ERR, _log_prefix, "error while requesting version negotiation: " .. err)
+      assert(ngx.timer.at(math.random(5, 10), function(premature)
+        self:communicate(premature)
```

Comment:
If CP doesn't understand the process of the handshake, unlikely it will understand next try.
How should we handle this error?

Reply:
Also, this is a to-do to check CP actually supports the clustering protocol. And before sending the request we can check whether there's a cached response in the shm.

Reply:
AFAICT, this is the same logic as on the current CP/DP connection, if it doesn't work, keep retrying.  my guess is that it makes it simpler to (re)start both sides without worrying about which one must be first.